dali/test/python/input/test_video.py

```diff
@@ -0,0 +1,187 @@
+    decoder_pipe.build()
+    decoder_pipe.feed_input(input_name, [encoded_video])
+    decoder_out = decoder_pipe.run()
+    return decoder_out[0].as_array()[0].shape[0]
+
+
+def portion_out_reference_sequence(decoder_pipe_out, frames_per_sequence, batch_size):
+    """
+    A generator, that takes the output from VideoDecoder DALI pipeline. Then, based of the
+    provided parameters, it serves sequences one-by-one, which are supposed to be returned
+    by VideoInput operator.
+    """
+    ref_sequence = decoder_pipe_out[0].as_array()[0]
+    num_frames = ref_sequence.shape[0]
+    n_batches = num_frames // (batch_size * frames_per_sequence)
+    ref_sequence = ref_sequence[:n_batches * frames_per_sequence * batch_size]
+    sh = ref_sequence.shape
+    ref_sequence = ref_sequence.reshape(
+        n_batches, batch_size, frames_per_sequence, sh[1], sh[2], sh[3])
+    for rs in ref_sequence:
+        yield rs
+
+
+@params(*list(params_generator()))
+def test_video_input_compare_with_video_decoder(device, frames_per_sequence, batch_size, test_file):
+    """
+    Compares the VideoInput with the VideoDecoder.
+    """
+    input_name = "VIDEO_INPUT"
+
+    decoder_pipe = video_decoder_pipeline(input_name=input_name, batch_size=1, device=device,
+                                          **common_pipeline_params)
+    input_pipe = video_input_pipeline(input_name=input_name, batch_size=batch_size,
+                                      frames_per_sequence=frames_per_sequence, device=device,
+                                      **common_pipeline_params)
+
+    decoder_pipe.build()
+    decoder_pipe.feed_input(input_name, [test_file])
+    decoder_out = decoder_pipe.run()
+
+    input_pipe.build()
+    input_pipe.feed_input(input_name, np.array([[test_file]]))
+
+    for ref_seq in portion_out_reference_sequence(decoder_out, frames_per_sequence, batch_size):
+        input_out = input_pipe.run()
+        test_seq = input_out[0].as_array()
+        assert np.all(ref_seq == test_seq)
+
+
+@params(*list(params_generator()))
+def test_video_input_partial_vs_pad(device, frames_per_sequence, batch_size, test_video):
+    input_name = "VIDEO_INPUT"
+    partial_pipe = video_input_pipeline(input_name=input_name, batch_size=batch_size,
+                                        frames_per_sequence=frames_per_sequence, device=device,
+                                        last_sequence_policy='partial', **common_pipeline_params)
+    pad_pipe = video_input_pipeline(input_name=input_name, batch_size=batch_size,
+                                    frames_per_sequence=frames_per_sequence, device=device,
+                                    last_sequence_policy='pad', **common_pipeline_params)
+
+    num_frames = get_num_frames(test_video)
+
+    partial_pipe.build()
+    partial_pipe.feed_input(input_name, np.array([[test_video]]))
+    pad_pipe.build()
+    pad_pipe.feed_input(input_name, np.array([[test_video]]))
```

Comment:
Do you ever test scheduling multiple files in the queue through number of feed_input calls?

Reply:
Good point. Added the test.

Reply:
Good point. I'll add a test